collect_app/src/main/java/org/odk/collect/android/mainmenu/MainMenuActivity.kt

```diff
@@ -90,6 +110,10 @@ class MainMenuActivity : LocalizedActivity() {
         initMapbox()
         initButtons()
         initAppName()
+
+        if (permissionsViewModel.shouldAskForPermissions()) {
```

Comment:
I think this kind of logic is probably ripe to be refactored into using Jetpack's Navigation component. That's kind of "all or nothing though" - probably not worth it for this issue, but something to keep in mind for the future.

Reply:
Maybe `PermissionsDialogFragment` should check it on its own and not call `.create()` if `permissionsViewModel.shouldAskForPermissions()` returns false?

Reply:
I did initially have this structure, but it was really difficult to test due to some missing parts of AndroidX test (see `@Ignore` tests in https://github.com/getodk/collect/pull/5590/commits/5909e4ef041450e8fa1a4430aa061c1466661440). It also felt simpler (and actually more efficient) to have the decision about showing the dialog outside.